### DISCUSSION TITLE
Tabs closes themselfs after resurrection

### DISCUSSION THREAD

**Discussion by explesy** [zellij-org/zellij#3694] [Q&A] [answered]
So for example lets say that I have a tab with open nvim. Then I'm doing reboot and resurrecting that session, then I press Enter and nvim starts. But when I'm trying to exit from nvim it shows me only black screen not my terminal promt and I can only press ctrl-c and that tab closes. And all this right for all terminal programs that I run, not only nvim.

Why is that so and how to fix it?

**Comment by Lockszmith-GH** [comment]
That's the session resurrection feature.

What happens is that the resurrection layout is tied to the nvim command session resurrection saved.
The reason you're seeing a blank pane is most likely because you hid the pane borders (hit <kbd>Ctrl</kbd>+<kbd>p</kbd> followed by <kbd>z</kbd> and you should see the session resurrection prompt in the border.
Hitting <kbd>ESC</kbd> will drop you to your shell, but the layout is still tied to the nvim command.

Documentation: https://zellij.dev/documentation/session-resurrection
#2919 - talks about how to stop resurrection.

**Reply by explesy** [reply]
yeah, I'm totally ok with resurrection and like this feature. I just don't understand why when I'm closing nvim it's doesn't show me my terminal but closes the whole tab. Is it how it should be or not?

**Reply by Lockszmith-GH** [reply, marked as answer]
Yes, this is by design - the session Resurrection saved that pane with the `nvim command line`, not with the shell running that command line.

The default behavior for a non-shell application that runs in a zellij pane is to wait and restart the command if any key is hit.
Think about a development environment where you want to test your app - it runs then exits, you want to keep the code editor on one pane and the execution on another - if upon exit, that pane closes, you can't see the output, so by default, zelilj assumes the output is important and stays suspended - waiting for any key to retry, <kbd>Ctrl</kbd>+<kbd>c</kbd> to close, or <kbd>ESC</kbd> to drop to the shell.

**Reply by explesy** [reply]
omg, now I understand that I can just press `ESC` to drop to the shell. Thank you so much!

**Reply by Lockszmith-GH** [reply]
I mentioned it earlier, but it probably slipped your attention:
<kbd>Ctrl</kbd>+<kbd>p</kbd>, <kbd>z</kbd> will show you the pane's border - you'll see the keyboard hints at the bottom border.
